### ISSUE TITLE
[Keyboard Navigation- Make code - Reset]: Focus doesn't land on triggered control after activating close button on dialog.

### ISSUE THREAD

**Issue by Yamini234u** [microsoft/pxt-microbit#5476]
### **User Experience:**
Keyboard dependent users will face difficulty if visual focus does not land back on triggered control element after closing the dialog, then user who rely on keyboard will not be able to predict the focus is on which control so the end users will get confused and this may lead to disorientation due to this issue.
**Note:**
User credentials should NOT be included in the bug.

### **Repro Steps:**

1.  Launch application URL : https://makecode.microbit.org
2. The home page will open.
3. Navigate "New Project" option and select it.
4. Editor screen will open.
5. Navigate to the "Settings" option appearing in header section and select it.
6. Navigate to "Reset" option and select it.
7. Navigate through all the controls appearing under "Reset" dialog.
8. Navigate to close button of dialog and press enter
9. Observe and verify whether focus lands on triggered control after closing the dialog.
10. ### **Actual Result:**
11. Issue: Focus doesn't land on triggered control after activating close button on dialog.

**Observation:**
On activating close button focus landing on "Header Part" user need press more than 8 keystrokes to reach the control.

### **Expected Result:**
Focus should land on triggered control after activating close button on dialog.
**Solution:**: After close button on dialog, focus should land on "Settings" control.

### **MAS Reference:**
[MAS 2.4.3 – Focus Order](https://aka.ms/MAS2.4.3)

**ARIA Authoring Ref Link/Fluent Patterns Guidance /HITS Guidance:**
N/A

**UIA/DOM details available:**
N/A
 ### **Reference Links:**

- [Accessibility Insights!](https://accessibilityinsights.io/) - Identify accessibility bugs before check-in and make bug fixing faster and easier.
- External Bug Process: If this bug belongs to an external team, mark it as resolved/done and assign it back to the tester with notes on where to file/route the bug. For more information, please use this [External Bug Process (sharepoint.com)](https://microsoft.sharepoint.com/teams/cea11y/SitePages/External-Bug-Process.aspx?xsdata=MDN8MDF8fDE1MThmM2JlZGFjMDRlZDhiZmViYTUxMDA2M2NiZWI3fDcyZjk4OGJmODZmMTQxYWY5MWFiMmQ3Y2QwMTFkYjQ3fDF8MHw2Mzc3NDEyNTM2MTEzNzIwNzR8R29vZHxWR1ZoYlhOVFpXTjFjbWwwZVZObGNuWnBZMlY4ZXlKV0lqb2lNQzR3TGpBd01EQWlMQ0pRSWpvaUlpd2lRVTRpT2lJaUxDSlhWQ0k2TVRKOQ%3D%3D&sdata=MnlpSXlVMnpGT3cvZ0xJNnFWUWZWTGt6dmo1VkJMZFpIZHBvUUduNDhyMD0%3D&ovuser=72f988bf-86f1-41af-91ab-2d7cd011db47%2cv-kgeetala%40microsoft.com). “HCL Staff should not log any third-party external bugs. Should be routed to EDAD team”.
- Please reach out to [C&AI Teams channel](https://apc01.safelinks.protection.outlook.com/?url=https://teams.microsoft.com/l/team/19%253a3cd071821b694f11b054d878453cd75f%2540thread.skype/conversations?groupId%3D30e3f968-15dd-4da2-bd8e-85f27e4af2c3%26tenantId%3D72f988bf-86f1-41af-91ab-2d7cd011db47&data=02%7c01%7cAklavyaM%40hcl.com%7c24e89c6ac26f42cfe40c08d7667e49f3%7c189de737c93a4f5a8b686f4ca9941912%7c0%7c0%7c637090564906844600&sdata=22JjBMoA1ID/KnYqFa/OB%2BrkTaLh5YYsGBd/7Bm7u5w%3D&reserved=0) for any process-related queries.

### **Test Environment:**
- **OS Version:** Windows 11 23H2 (OS Build:22631.3007)
- **Browse**r: Chrome Version 121.0.2277.4 (Official build) dev (64-bit).
- **Test Matrix:** Chrome + JAWS
- **URL:** https://makecode.microbit.org/.

**Video Attachment**:https://github.com/microsoft/pxt-microbit/assets/117158108/ed8c4933-50dc-4d3e-b754-e3742407494a

**Comment by Yamini234u** [comment]
#Accessibility;#A11yMAS;#A11ySev2;#AILimited;#MAS2.4.3;#HCL;#Nonbenchmark;#Win11-Chrome;#E2E-HCL-MakeCode-Jan2024

**Comment by Yamini234u** [comment]
#Accessibility;#A11yMAS;#A11ySev2;#AILimited;#k4W;#MAS2.4.3;#HCL;#Nonbenchmark;#Win11-Chrome;#E2E-HCL-MakeCode-Jan2024

**Comment by Yamini234u** [comment]
#Accessibility;#A11yMAS;#A11ySev2;#AILimited;#k4W;#MAS2.4.3;#HCL;#Nonbenchmark;#Win11-Chrome;#E2E-HCL-MakeCode-Jan2024
